testing-chocolatey/local-testing.md

```diff
@@ -0,0 +1,86 @@
+
+[vagrant-download]: https://www.vagrantup.com/downloads
+[vbox-download]: https://www.virtualbox.org/wiki/Downloads
+
+## Getting Started
+
+A `Vagrantfile` is provided for ease of use, which simplifies the necessary setup.
+If you would like to setup your own VMs or servers for the purposes of this testing, see the [Generic](#generic) instructions.
+
+Otherwise, follow the [Vagrant](#vagrant) instructions.
+
+Once your VMs or servers are ready, follow the instructions in [Running `ansible-test`](#running-ansible-test) to run through the collection tests.
+
+### Vagrant
+
+1. Download the nupkg containing the Chocolatey version you'd like to test.
+1. Place the nupkg in the `provision_files` folder.
+1. Run `vagrant up` to spin up the Ubuntu 'server' and Windows 10 'client' VMs.
+
+Continue on to the [Running `ansible-test`](#running-ansible-test) section.
+
+### Generic
+
+1. Before continuing, prepare a Windows VM and a Linux VM (these instructions assume you're using Ubuntu 18.10 or newer, and a Windows 10 client)
+1. Run the following commands on the Linux server to setup Ansible and install the Chocolatey collection:
+
+    ```sh
+    sudo apt-get update
+    sudo apt-get install python3-venv -y
+
+    python3 -m venv ~/ansible
+    source ~/ansible/bin/activate
+
+    pip3 install --upgrade wheel
+    pip3 install ansible pywinrm
+
+    ansible-galaxy collection install chocolatey.chocolatey
+    ```
+
+1. Copy the **nupkg** file containing the version of Chocolatey you'd like to test onto the Windows VM, under `C:\packages` (you will probably need to create this folder as well).
+1. Run the following commands in an administrative session of Windows PowerShell to install Chocolatey and update it to the version you've placed in `C:\packages`:
+
+    ```powershell
+    & ([scriptblock]::Create((Invoke-WebRequest -Uri "https://chocolatey.org/install.ps1" -UseBasicParsing)))
+    choco upgrade chocolatey -y -s C:\packages --pre
+    ```
+
+1. Follow the instructions from [this blog post](https://pureinfotech.com/set-static-ip-address-windows-10/) to assign the Windows VM the static IP address `10.0.0.11`
+   This IP address is hard-coded into the `vagrant-inventory.winrm` file found in the `~/.ansible/collections/ansible_collections/chocolatey/chocolatey/tests/integration/targets` folder.
+   If you prefer, you can modify that file in your Linux server to point to a different IP address instead.
+
+## Running `ansible-test`
+
+1. If using Vagrant, SSH into the Ansible server with the following command:
+
+    ```sh
+    vagrant ssh choco_ansible_server
+    ```
+
+   Otherwise, open your Linux VM directly and continue from the VM terminal.
+1. Run the chocolatey.chocolatey collection tests against the Windows client machine:
+
+    ```sh
+    source ~/ansible/bin/activate
+    cd ~/.ansible/collections/ansible_collections/chocolatey/chocolatey
+    ansible-test windows-integration -vvvv --inventory vagrant-inventory.winrm --requirements
```

Comment:
What are the expected outcomes from running this command?

I saw lots of blue, green, and yellow text (don't think I saw any red) and at the end, I got this output:

![image](https://user-images.githubusercontent.com/1271146/117427332-d99fb680-af1c-11eb-84bb-f4bc73ef558f.png)

Does this mean it worked?

Reply:
yup!

Yeah, the ansible-test output is... not the _most_ clear to read through. If you get errors, there'll be failures noted in that last result. And then it's a bit of sifting through some of the prior logs to find the details.

Thus far I've not managed to get much clearer output for the integration tests when they fail. If they succeed, they can output XML reports -- but from what I've gathered setting up the CI pipeline which uses the same tests, for some reason I cannot fathom the XML report with the nice output doesn't seem to be generated when something fails.

Test reports should be generated to the `./tests/output/` folder (relative to the `chocolatey/chocolatey` collection folder).